cmake/BuildTests.cmake

```diff
@@ -151,11 +151,13 @@ SET(INT_TEST_COUNT 0)
 FOREACH(testfile ${INTEGRATION_TESTS})
   get_filename_component(testfilename "${testfile}" NAME_WE)
   add_executable("${testfilename}" "${TEST_DIR}/integration/${testfile}")
+  target_wholearchive_library("${testfilename}" minifi-standard-processors)
```

Comment:
What changed that makes this link necessary?

Reply:
UnorderedMapKeyValueStoreService (or some other state storage) has to be registered, otherwise we can't use it. Registration is done at static initialization time using the REGISTER_RESOURCE macro. It would be optimized out if not linked and test would fail.

Reply:
And do we need whole archive, wouldn't normal linking be enough? I think static initialization would be done even with normal linking, but I'm not 100% sure.

Reply:
My experience is that it would be optimized out. Also seen this in another project. Without this line the test would fail (it did until I came up with this). Maybe it also depends on linkers and their settings.
https://stackoverflow.com/questions/7327741/initialisation-of-static-object-when-linking-against-a-static-library

Reply:
I think we have UnorderedMapKeyValueStoreService working on main without this change. I wonder how was it working before this change?

Reply:
In which executable? This one affects integration tests.

Reply:
The same whole archive linking is done for unit tests of standard processors, nanofi tests and many other tests.
The main minifiexe binary is also linking it through linking all the extensions (including standard processors) in main/CMakeLists.txt:72:
`target_wholearchive_library(minifiexe ${EXTENSION})`

Reply:
Line was moved after discussion in private.